lib/travis/build/script/csharp.rb

```diff
@@ -20,19 +22,30 @@ def configure
 
 
           sh.fold('mono-install') do
-            if config[:csharp] == 'mono'
+            if mono_version_valid?
               sh.echo 'Installing Mono', ansi: :yellow
 
-              sh.cmd 'sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF', echo: false
-              sh.cmd "sudo sh -c \"echo 'deb http://download.mono-project.com/repo/debian wheezy main' >> /etc/apt/sources.list.d/mono-xamarin.list\"", echo: false
-              sh.cmd "sudo sh -c \"echo 'deb http://download.mono-project.com/repo/debian wheezy-libtiff-compat main' >> /etc/apt/sources.list.d/mono-xamarin.list\"", echo: false
-              sh.cmd 'sudo apt-get update -qq', timing: true
-              sh.cmd 'sudo apt-get install -qq mono-complete referenceassemblies-pcl nuget mono-vbnc fsharp', timing: true
-              sh.cmd 'mozroots --import --sync --quiet', timing: true
+              if !is_mono_2?
+                sh.cmd 'sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF', echo: false
+                sh.cmd "sudo sh -c \"echo 'deb http://download.mono-project.com/repo/debian #{mono_repo} main' >> /etc/apt/sources.list.d/mono-xamarin.list\"", echo: false
+                sh.cmd "sudo sh -c \"echo 'deb http://download.mono-project.com/repo/debian wheezy-libtiff-compat main' >> /etc/apt/sources.list.d/mono-xamarin.list\"", echo: false
+                sh.cmd 'sudo apt-get update -qq', timing: true
+                sh.cmd 'sudo apt-get install -qq mono-complete referenceassemblies-pcl nuget mono-vbnc fsharp', timing: true
+                sh.cmd 'mozroots --import --sync --quiet', timing: true
+              else
+                sh.cmd 'sudo apt-get install -qq mono-complete mono-basic', timing: true
```

Comment:
there's no `mono-basic` package, the one on launchpad you're referring too is just the name of the source, the package name is still `mono-vbnc`: http://bazaar.launchpad.net/~ubuntu-branches/ubuntu/vivid/mono-basic/vivid/view/head:/debian/control#L70

> People who are fetching nuget as part of their buildpacks are probably running mozroots already and it wastes time for those who aren't.

You could apply the same argument to the newer version as well, we install nuget+mozroots even though not everyone might be using it. I'd prefer playing safe and make it consistent in favor of saving a few seconds of build time.